src/components/AxisCollection/CollapsedAxis.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react';
+import * as d3 from 'd3';
+import PropTypes from 'prop-types';
+import isEqual from 'lodash.isequal';
+import { createYScale } from '../../utils/scale-helpers';
+import { singleSeriePropType } from '../../utils/proptypes';
+
+export default class CollapsedAxis extends Component {
+  static propTypes = {
+    // TODO: Zooming is currently not supported.
+    // It might be nice to support zooming all of the axes when you zoom on
+    // the collapsed one. Experiment with this and either add it or remove
+    // all of this unused code.
+    zoomable: PropTypes.bool,
+    series: PropTypes.arrayOf(singleSeriePropType),
+    height: PropTypes.number.isRequired,
+    width: PropTypes.number.isRequired,
+    updateYTransformation: PropTypes.func,
+    yTransformation: PropTypes.shape({
+      y: PropTypes.number.isRequired,
+      k: PropTypes.number.isRequired,
+      rescaleY: PropTypes.func.isRequired,
+    }),
+    color: PropTypes.string,
+  };
+
+  static defaultProps = {
+    series: [],
+    zoomable: false,
+    updateYTransformation: () => {},
+    yTransformation: null,
+    color: '#666',
+  };
+
+  componentWillMount() {
+    this.zoom = d3.zoom().on('zoom', this.didZoom);
+  }
+
+  componentDidMount() {
+    this.selection = d3.select(this.zoomNode);
+    this.syncZoomingState();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.zoomable !== this.props.zoomable) {
+      this.syncZoomingState();
+    }
+    if (this.props.yTransformation) {
+      if (!isEqual(prevProps.series.yDomain, this.props.series.yDomain)) {
+        this.selection.property('__zoom', this.props.yTransformation);
+      }
+    }
+  }
+
+  syncZoomingState = () => {
+    if (this.props.zoomable) {
+      this.selection.call(this.zoom);
+    } else {
+      this.selection.on('.zoom', null);
+    }
+  };
+
+  didZoom = () => {
+    const t = d3.event.transform;
+    this.props.updateYTransformation(
+      this.props.series.id,
+      t,
+      this.props.height
+    );
+  };
+
```

Comment:
Having a strong feeling we could move common with YAxis component functionality into a separate component or a factory.

Reply:
Yeah, this is pretty hideous -- I want to fix all of this pixel-math in a separate PR.